hr_payroll_cancel/models/hr_payroll.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# Copyright 2014 - Vauxoo http://www.vauxoo.com/
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# Copyright 2017 Serpent Consulting Services Pvt. Ltd.
+#   (<http://www.serpentcs.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
+
+
+class HrPayslip(models.Model):
+    _inherit = 'hr.payslip'
+
+    @api.multi
+    def action_payslip_cancel(self):
+        if self.credit_note:

```

Comment:
The credit_note flag is activated in the refund not in the original payslip. I am still able to cancel a payslip that can be refunded.